src/vue/directives/tooltip.js

```diff
@@ -0,0 +1,387 @@
+    tooltipArrow.style.borderBottom = `8px solid ${bgColor}`
+    tooltipArrow.style.borderTop = '0'
+    tooltipArrow.style.borderRadius = '0'
+    tooltipArrow.style.position = 'absolute'
+    const twRect = tooltipWrapper.getBoundingClientRect()
+    const targetRect = target.getBoundingClientRect()
+    const startPointX = targetRect.left - twRect.left
+    tooltipArrowRect = tooltipArrow.getBoundingClientRect()
+    switch (currentPosition) {
+      case DIRECTIONS.top:
+        tooltipArrow.style.bottom = '-8px'
+        tooltipArrow.style.transform = 'rotate(180deg)'
+        tooltipArrow.style.left = startPointX + (targetRect.width / 2) -
+          (tooltipArrowRect.width / 2) + 'px'
+        break
+      case DIRECTIONS.bottom:
+        tooltipArrow.style.top = '-8px'
+        tooltipArrow.style.left = startPointX + (targetRect.width / 2) -
+          (tooltipArrowRect.width / 2) + 'px'
+        break
+      case DIRECTIONS.left:
+        tooltipArrow.style.right = '-12px'
+        tooltipArrow.style.transform = 'rotate(90deg)'
+        break
+      case DIRECTIONS.right:
+        tooltipArrow.style.left = '-12px'
+        tooltipArrow.style.transform = 'rotate(-90deg)'
+        break
+    }
+  }
+
+  // checks if any other tooltips are open
+  function detectOpenedTooltips () {
+    const listener = (ev) => {
+      if (ev.target.closest('.tooltip-container')) return
+      removeTooltip()
+      currentTarget = null
+      tooltipOn = false
+      document.removeEventListener('click', listener)
+    }
+    document.addEventListener('click', listener)
+  }
+
+  function removeTooltip () {
+    if (tooltipWrapper.parentNode) {
+      tooltipWrapper.parentNode.removeChild(tooltipWrapper)
+      currentTarget.classList.remove('tooltip-container')
+    }
+  }
+
+  return {
+    bind (el, binding) {
+      bindedOpts = binding
+      if (bindedOpts.value.type === 'click') {
+        el.addEventListener('click', (e) => {
+          currentTarget = currentTarget || e.target
+          tooltipOn = !tooltipOn
+          if (currentTarget !== e.target) {
+            currentTarget = e.target
+            removeTooltip()
+            tooltipOn = true
+          }
+          detectOpenedTooltips()
+          resizeHandler = renderTooltip.bind(null, e, binding)
+          if (tooltipOn) {
+            renderTooltip(e, binding)
+            window.addEventListener('resize', function () {
+              removeTooltip()
+              resizeHandler()
+            })
+          } else {
+            removeTooltip()
+          }
+        })
+      } else {
+        el.addEventListener('mouseover', (e) => {
+          currentTarget = currentTarget || e.target
+          if (currentTarget !== e.target) {
+            currentTarget = e.target
+            removeTooltip()
+            tooltipOn = false
+          }
+          renderTooltip(e, binding)
+        })
+
+        el.addEventListener('mouseout', (e) => {
```

Comment:
Check that all events are destroyed after directive unbind

Reply:
Done